### ISSUE TITLE
False positive for unnecessary parentheses

### ISSUE THREAD

**Issue by koromorix** [DotNetAnalyzers/StyleCopAnalyzers#2466]
**Code**:

```cs
static bool isString(object o) {
	if (!(o is string s)) return false;
	return true;
}
```

**Expected**: No error.
**Actual**: `Program.cs(5,8): warning SA1119: Statement must not use unnecessary parenthesis`

**Comment by sharwell** [comment]
This appears to be a duplicate of #2372, which was fixed for the 1.1.0-beta004 release. If you are using this release (or newer) and still encounter the behavior, please reopen this issue.
